### ISSUE TITLE
SA1003 conflicts with SA1010 for new style collection initialisation.

### ISSUE THREAD

**Issue by mrudat** [DotNetAnalyzers/StyleCopAnalyzers#3843]
I think this is the preferred option:

```cs
    /// <summary>
    /// Gets or sets the list of <see cref="Thing"/>s.
    /// </summary>
    [SuppressMessage("StyleCop.CSharp.SpacingRules", "SA1010:Opening square brackets should be spaced correctly", Justification = "Conflicts with SA1003")]
    public Thing[] Things { get; set; } = [];
```

```cs
    /// <summary>
    /// Gets or sets the list of <see cref="Thing"/>s.
    /// </summary>
    [SuppressMessage("StyleCop.CSharp.SpacingRules", "SA1003:Symbols should be spaced correctly", Justification = "Conflicts with SA1010")]
    public Thing[] Things { get; set; } =[];
```

**Comment by mrudat** [comment]
Also hit by the same conflicting rules:
```cs
var thing = new Thing(){
   Collection = []
}
```

**Comment by sharwell** [comment]
Duplicate of #3687
